File: LayoutTests/http/tests/navigation/resources/check-ping.php
<?php
while (!file_exists("ping.txt")) {
    usleep(10000);
    // file_exists() caches results, we want to invalidate the cache.
    clearstatcache();
}

echo "<html><body>\n";
echo "Ping sent successfully";
$pingFile = fopen("ping.txt", 'r');
while ($line = fgets($pingFile)) {
    echo "<br>";
    echo trim($line);
}
fclose($pingFile);
unlink("ping.txt");
echo "<script>";
echo "if (window.testRunner)";
echo "    testRunner.notifyDone();";
echo "</script>";
echo "</body></html>";
?>

File: LayoutTests/http/tests/navigation/resources/check-ping.py
#!/usr/bin/env python3

import os
import sys
import time
from ping_file_path import ping_filepath

sys.stdout.write('Content-Type: text/html\r\n\r\n')

iterations = 100
while not os.path.isfile(ping_filepath) and iterations != 0:
    time.sleep(0.01)
    iterations -= 1

if iterations == 0:
    sys.stdout.write(
        '<html><body>\n'
        'Ping not sent'
    )
else:
    sys.stdout.write(
        '<html><body>\n'
        'Ping sent successfully'
    )

    with open(ping_filepath, 'r') as ping_file:
        for line in ping_file.readlines():
            sys.stdout.write('<br>{}'.format(line.strip()))

    if os.path.isfile(ping_filepath):
        os.remove(ping_filepath)

sys.stdout.write(
    '<script>'
    'if (window.testRunner)'
    '    testRunner.notifyDone();'
    '</script>'
    '</body></html>'
)
